### PR TITLE
Apply ocamlformat to backend/cfg

### PR DESCRIPTION
**Pull request by gretay-js**
Extend "make fmt" and "make check-fmt" targets to format files in the `backend/cfg` directory.

Move  .ocamlformat from middle_end/flambda2 to the top level of the repo. This would allow us to format new files we add in `backend` subdirectory. One potential problem is that if an editor is set to format when a file is saved, the new location of .ocamlformat will cause all files in the repo to be formatted, but I think it can be controlled by adding .ocamlformat-ignore file.

There is a separate commit to reformat all the files.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -34,10 +34,11 @@ We may then use a git submodule to import it into the Flambda backend repo.
 
 ## Code formatting for Flambda 2
```
Comment (mshinwell) [inline HACKING.md:35]: Remove " for Flambda 2"

File: HACKING.md [hunk 2]
```diff
@@ -34,10 +34,11 @@ We may then use a git submodule to import it into the Flambda backend repo.
...
 
-The CI checks that all Flambda 2 code (in `middle_end/flambda2/`) is
+The CI checks that all Flambda 2 code (in `middle_end/flambda2/`) and
+Cfg code (in `backend/cfg`) is
```
Comment (mshinwell) [inline HACKING.md:38]: Trailing slash after "cfg" for consistency

File: Makefile.in [hunk 3]
```diff
@@ -489,16 +489,21 @@ fmt:
...
 	    -and \! \( -name "flambda_parser.*" -or -name "flambda_lex.*" \))
+	ocamlformat -i \
+	  $$(find backend/cfg \
+	    \( -name "*.ml" -or -name "*.mli" \))
 
 .PHONY: check-fmt
 check-fmt:
-	if [ "$$(git status --porcelain middle_end/)" != "" ]; then \
+	if [ "$$(git status --porcelain middle_end/)" != "" ] || \
+           [ "$$(git status --porcelain backend/)" != "" ]; then \
```
Comment (mshinwell) [inline Makefile.in:499]: This should say `backend/cfg/`, no?  Same below.

**Review by mshinwell — CHANGES_REQUESTED** [review]

File: HACKING.md (hunk 1 above)
Comment (gretay-js) [reply HACKING.md]: fixed.

File: HACKING.md (hunk 2 above)
Comment (gretay-js) [reply HACKING.md]: fixed.

File: Makefile.in (hunk 3 above)
Comment (gretay-js) [reply Makefile.in]: Fixed both cfg and flamdba2 case here and below.

**Review by mshinwell — APPROVED** [review]

**Comment by mshinwell** [comment]
I'll merge this once CI passes, we don't need to bother the normal file reviewers.

**Comment by hhugo** [comment]
I'm surprised no to see an ocamlformat version specified in the config (. ocamlformat). It is usually considered a good practice to avoid reformatting the whole codebase when a different version of ocamlformat is used.

**Comment by gretay-js** [comment]
> I'm surprised no to see an ocamlformat version specified in the config (. ocamlformat). It is usually considered a good practice to avoid reformatting the whole codebase when a different version of ocamlformat is used.

I didn't know about versioning, thank you! I'll fix in a separate PR.

**Pull request merged**